### PR TITLE
Upgrade Azure SDK

### PR DESCRIPTION
**Pull request by fcofdez**

### CODE REVIEW

File: modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java [hunk 1]
```diff
@@ -276,4 +284,22 @@ private void trackSuccessfulRequest(HttpRequest httpRequest, HttpResponse httpRe
...
+    // to avoid creating multiple connection pools.
+    record NioLoopResources(EventLoopGroup eventLoopGroup) implements LoopResources {
+        @Override
+        public EventLoopGroup onServer(boolean useNative) {
+            // We rely on Nio here, hence we ignore the useNative flag
+            return eventLoopGroup;
+        }
+    }
+
+    static class Resolver extends AddressResolverGroup<InetSocketAddress> {
```
Comment (arteam) [inline modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java:299]: Isn't this resolver is the same as `DefaultAddressResolverGroup.INSTANCE`?

**Review by arteam — COMMENTED** [review]
LGTM! I've left a couple of stylistic comments

File: modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java (hunk 1 above)
Comment (fcofdez) [reply modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java:299]: You're right, this is a leftover from when I was tracking down the issue with the connection pool 👍

**Review by arteam — APPROVED** [review]
LGTM! Thanks Francisco!

**Comment by weizijun** [comment]
hi, @fcofdez @arteam , is the PR ok to merged, it will fix jaskson bad case, releated https://github.com/elastic/elasticsearch/pull/80160

**Comment by arteam** [comment]
Ping @fcofdez, I think we are good to go to merge this PR.

**Comment by fcofdez** [comment]
ops! This fell through the cracks! I'll merge it!

**Comment by fcofdez** [comment]
I have to fix https://github.com/elastic/elasticsearch/issues/84231 before merging this one (it's related to the test infrastructure)

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by jkakavas** [comment]
@fcofdez any chance to merge this soon so that it can make the next release ? I see https://github.com/elastic/elasticsearch/issues/84231 is resolved

**Comment by rjernst** [comment]
@fcofdez The example-plugins CI check failure can be ignored for now, it is due to the recent version bump. This PR should be good to merge. I think it should go back to 8.2 and 7.17.

**Comment by weizijun** [comment]
hi, @fcofdez , I see you are revert the jackson version. Can you tell what's the revert reason? There is a bug in jackson, and need to be upgrade. Here is the close issue: https://github.com/elastic/elasticsearch/pull/80160

**Comment by fcofdez** [comment]
Jackson has been upgraded separately in this PR https://github.com/elastic/elasticsearch/pull/86051 cc @weizijun

**Comment by weizijun** [comment]
> Jackson has been upgraded separately in this PR #86051 cc @weizijun

Yeah, Nik told me about this PR, thanks!

**Comment by nik9000** [comment]
Ah! I should have LinkedIn that publicly! I'm slipping.

**Pull request merged**